lib/puppet/defaults.rb

```diff
@@ -518,6 +518,12 @@ def self.default_diffargs
       :desc    => "Stores trusted node data in a hash called $trusted.
         When true also prevents $trusted from being overridden in any scope.",
     },
+    :trusted_server_facts => {
+      :default => false,
+      :type    => :boolean,
+      :desc    => "Stores a trusted set of server-side global variables in a hash
+        called $server_facts, which cannot be overridden by client facts.",
```

Comment:
"...cannot be overridden by client_facts or logic in manifests. Makes it illegal to assign to the variable $server_facts in any scope."